automation/src/pages/HomePage.ts

```diff
@@ -0,0 +1,42 @@
+import {ElementHandle} from 'puppeteer-core';
+import SignedInPage from './SignedInPage';
+
+/**
+ * xpath selectors.
+ */
+const selectors = {
+  pageTitle: 'Homepage',
+  header: '//h3[normalize-space(text())=\'Workspaces\']',
+  seeAllWorkspacesLink: '//*[normalize-space()=\'See all Workspaces\']'
+};
+
+export default class HomePage extends SignedInPage {
+
+  public async isLoaded(): Promise<void> {
+    await super.isLoaded();
+    await this.waitUntilDocumentTitleMatch(selectors.pageTitle);
+    await this.puppeteerPage.waitForXPath(selectors.seeAllWorkspacesLink, {visible: true});
+    await this.puppeteerPage.waitForXPath(selectors.header, {visible: true})
+  }
+
+  public async waitForReady(): Promise<HomePage> {
+    await this.isLoaded();
+    await this.waitForSpinner();
+    return this;
+  }
+
+  public async goto(): Promise<HomePage> {
+    await this.navigation.toHome();
+    return this;
+  }
+
+  get seeAllWorkspacesLink(): Promise<ElementHandle> {
+    return this.puppeteerPage.waitForXPath(selectors.seeAllWorkspacesLink, {visible: true})
+  }
+
+  public async toAllWorkspaces() {
+    const link = await this.seeAllWorkspacesLink;
+    await Promise.all([link.click(), this.puppeteerPage.waitForNavigation({waitUntil: 'networkidle0'})]);
```

Comment:
Is it necessary to explicitly wait for navigation? If we just wait for the next interaction directly, it should cross page boundaries without having to do that manually.